### ISSUE TITLE
Hands on project

### ISSUE THREAD

**Issue by tjwei** [pycourse/inumpy#2]
Any suggestion on the list of hands on problems?
Image recognition sees fun, but does that mean we need scikit-image and scikit-learn?
Or build something from scratch?

**Comment by tjwei** [comment]
JPEG-like DCT image compression  is also a good hands-on experiment. it is visual, easy and only depends on numpy and matplotlib.

**Comment by yungyuc** [comment]
We should build things from scratch.  We need to make the code very simple.  The example code doesn't need to be reusable

scikit-image and scikit-learn are good topics.  They should be in an additional workshop.

Image compression is a good example.

**Comment by tjwei** [comment]
Another visual hands on task is to make a hybrid image from two, like Marilyn Einstein. It is also very easy and only depends on numpy and matplotlib.

**Comment by tjwei** [comment]
@yungyuc Agreed, so we shall choose an image recognition algorithm that is not too complicated.

**Comment by yungyuc** [comment]
That's a good example (Marilyn Einstein), and I want to learn it myself too.  I think all instructors should get familiar with the code before the workshop.

**Comment by yungyuc** [comment]
@tjwei Do you mind just write the discussions into the syllabus file?  I think once we have an agreement on something we can right away to push it.  No need to wait.  If for some subjects further discussion is preferred, we can make it a PR and discuss there.

**Comment by tjwei** [comment]
No problem.

**Comment by tjwei** [comment]
Create a hands-on-example branch and pull request #6 . Move discussion to the pull request. 
Can edit README.md online under this branch before we want to merge it into the master.

**Comment by alberthuang2718** [comment]
Should we add a simple hands on project for ipython+simplest numpy? So as to let students get used to ipython and their environments before going into more advanced parts?

**Comment by yungyuc** [comment]
@jjhuang314 Good idea.  What the subject of the project you have in mind?
